File: algorithms/[71]简化路径/solution.ts
// topics = ["栈"]

function simplifyPath(path: string): string {
  const st: string[] = [];

  for (const s of path.split('/')) {
    if (s == '' || s == '.') {
      continue;
    }
    if (s == '..') {
      if (st.length) {
        st.pop();
      }
    } else {
      st.push(s);
    }
  }

  return `/${st.join('/')}`;
}

File: algorithms/[71]简化路径/solution.py
# topics = ["栈"]

from typing import List


class Solution:
    def simplifyPath(self, path: str) -> str:
        st: List[str] = []

        for s in path.split('/'):
            if not s or s == '.':
                continue
            if s == '..':
                if st:
                    st.pop()
            else:
                st.append(s)

        return f'/{"/".join(st)}'
